crypto/cryptlib.c

```diff
@@ -401,8 +401,11 @@ void *OPENSSL_stderr(void)	{ return stderr; }
 int CRYPTO_memcmp(const void *in_a, const void *in_b, size_t len)
 	{
 	size_t i;
-	const unsigned char *a = in_a;
-	const unsigned char *b = in_b;
+	/* Must use "pointers to volatile" to tell the compiler to never
+	 * attempt optimizing the reads. Otherwise the compiler is allowed
+	 * to use LTO and convert this function into canonical version. */
+	const volatile unsigned char *a = in_a;
```

Comment:
Evidently this should be `const volatile unsigned char * volatile a = in_a;` and `const volatile unsigned char * volatile b = in_b;`.